components/ide/theia/leeway.Dockerfile

```diff
@@ -18,53 +17,73 @@ RUN mkdir -p /theia/node/bin \
 
 FROM alpine:3.9 AS builder_alpine
 
-RUN apk add --no-cache bash gcc g++ make pkgconfig python libc6-compat libexecinfo-dev git patchelf findutils
+RUN apk add --no-cache bash gcc g++ make pkgconfig python libc6-compat libexecinfo-dev git patchelf findutils curl jq
 
 # install node
 COPY --from=node_installer /theia/node/ /theia/node/
 ENV PATH=$PATH:/theia/node/bin/
 
+
 # install yarn by download+unpack to ensure it does NOT put anything into /theia/node/
 RUN wget https://github.com/yarnpkg/yarn/releases/download/v1.15.2/yarn-v1.15.2.tar.gz
 RUN tar zvxf yarn-v1.15.2.tar.gz
 ENV PATH=$PATH:/yarn-v1.15.2/bin/
 
-COPY components-theia-app--installer /theia-installer
+# yq - jq for YAML files
+RUN cd /usr/bin \
+    && curl -fsSL https://github.com/mikefarah/yq/releases/download/2.4.0/yq_linux_amd64 > yq \
+    && chmod +x yq
+ENV PATH=$PATH:/usr/bin/
+
+
 WORKDIR /theia
-RUN /theia-installer/install.sh
 
-# copy native dependencies of node
-COPY package-libs.sh /usr/bin/
-RUN package-libs.sh /theia/node/bin/node
+ENV THEIA_APP_COMMIT 90b4ba0b4c32222c7ef0946ffc30c7937e203777
+RUN mkdir theia-app \
+    && cd theia-app \
+    && git init \
+    && git remote add origin https://github.com/gitpod-io/theia-app \
+    && git fetch origin $THEIA_APP_COMMIT \
+    && git reset --hard FETCH_HEAD
+
+
+WORKDIR /theia/theia-app
+RUN yarn
+RUN yarn build
 
 # patching plugin ext host process to fix css extension compatibility, see https://github.com/gitpod-io/gitpod/pull/2483
-RUN cp -r /theia/node_modules/@gitpod/gitpod-ide/patches /theia/patches 
 RUN yarn patch-package
 
+
+# download builtin plugins
+RUN cd /theia/theia-app \
+    && sh ./scripts/builtin-plugins.sh
+
+
+# copy native dependencies of node
+COPY package-libs.sh /usr/bin/
+RUN package-libs.sh /theia/node/bin/node
+
 # copy native dependencies of node modules
-RUN find /theia/node_modules/ -iname *.node -exec package-libs.sh {} \;
+RUN find /theia/theia-app/node_modules/ -iname *.node -exec package-libs.sh {} \;
 
 RUN cp /theia/node/bin/node /theia/node/bin/gitpod-node && rm /theia/node/bin/node
 
+
 # cli config
 COPY bin /ide/bin
 RUN chmod -R ugo+x /ide/bin
 
 FROM scratch
 COPY --from=builder_alpine /theia/ /theia/
 
+ENV GITPOD_ENV_SET_GITPOD_BUILT_IN_PLUGINS /theia/theia-app/plugins/
```

Comment:
@akosyakov, that did work eventually :-)

Reply:
test